adapt-and-extend/theming/resourceregistry.rst

```diff
@@ -124,166 +236,258 @@ In ``mockup/patterns/structure/js/views/actionmenu.js``:
+See below for :ref:`some examples <bundle_example_records>` of different types of resource records.
 
-When developing an add-on you might want to create your own bundle. Alternatively, you can register your add-on code to be included in Plone's default ``plone`` bundle.
+When developing an add-on you will create your own bundle.
+Your bundle should include all resources required for your JavaScript or CSS/Less to work properly.
 
-If you need a bundle for a single page, you can define an extra bundle and only include it only there. The ``resourceregistry`` bundle for example is only used for the ``@@resourceregistry-controlpanel`` view. (see the section `Adding or removing bundles from a request`_ for more information)
+If your bundle will be used only on a single page, you can define it to include it only there.
+You can use the "expression" option to control including an enabled bundle.
+You can also use API methods from ``Products.CMFPlone.resources`` to add disabled bundles to a single request.
+For example, the ``resourceregistry`` bundle is only used for the ``@@resourceregistry-controlpanel`` view.
+(see the section :ref:`bundles_request_api` for more information)
 
 .. note::
 
-    A bundle can depend on another one.
-    This is mainly used for the order of inclusion in the rendered content and mostly relevant for legacy bundles.
-    Currently, bundle dependencies don't make use of RequireJS dependencies.
-    This means each bundle gets all of their dependencies compiled in, even if it was already used for another bundle.
+    A bundle can depend on other bundles.
+    Declaring such a dependency only controls the order in which bundles are loaded and is mostly relevant for legacy bundles.
+    Currently, bundle dependencies don't make use of RequireJS dependencies or AMD.
+    Each bundle will be compiled with all dependencies, even if a dependency was already used for another bundle.
     This raises the response payload unnecessarily.
+
     To avoid this, add your resources to existing bundles, like the "plone" bundle for production sites in your integration package.
-    You should still provide a custom bundle, so that users can see the JavaScript and CSS in action without having to revisit the resource registry and press the "Build" button.
+    Even if you do this, however, you should still provide a custom bundle containing only your code.
+    Then users can see the JavaScript and CSS for your package in action without having to revisit the resource registry and press the "Build" button.
 
-Bundle Definition
------------------
+Development vs. Production Mode
+-------------------------------
 
-Example based on Plone's standard bundles defined in ``Products/CMFPlone/profiles/dependencies/registry.xml``
+In development mode, each bundle loads all of its resources in the rendered site as individually, with individual requests.
```

Comment:
``... as individually, with individual requests`` maybe my english is not sufficient, but i dont get what that means exactly.